### ISSUE TITLE
Pathway analysis

### ISSUE THREAD

**Issue by KristofferSandas** [KristofferSandas/methods_seminar#5]
Lets see if it works to discuss here.

**Comment by sinmojito** [comment]
I have now created a local branch and will send an updated version of the pathway script when I am ready. It won't give me an option to send a pull request until I have modified the file anyways.

**Comment by KristofferSandas** [comment]
Right. Sounds good. Im sure we will find more things to write so we'll have a lot of good screenshots

**Comment by KristofferSandas** [comment]
This seems to work ok. Not as good as whatsapp, but it will have to do

**Comment by KristofferSandas** [comment]
New one should be the one I sent with both positive and negative logFC

**Comment by sinmojito** [comment]
# filter on min log2fold change (log2FoldChange > 2)
genes <- names(genes)[abs(genes) > 2]

I did some troubleshooting. the names(genes) part is causing the df to go NULL

**Comment by sinmojito** [comment]
I dunno why that first part was formatted as a heading but ok...

**Comment by sinmojito** [comment]
I think I see the issue. In the excel file, the A column needs to be the names for the p-values since those symbols will later be converted to ENTREZIDs with the bitr function.

**Comment by KristofferSandas** [comment]
Everything should be merged into main now, I'll have a look at the code.

**Comment by KristofferSandas** [comment]
Ah, the excel file is already filtered on FDR (adjusted P value) < 0.05 and abs(logFC)> 1.3. So we can run the whole thing through the anlysis. Something has broken the R package though. It can no longer load. I'll look into  it.

**Comment by KristofferSandas** [comment]
And I started working in the main branch now and then realized I need to switch to a new branch, so we'll see how this goes.. Worst case we'll start yet another R project.

**Comment by sinmojito** [comment]
i am still getting enrichGO error - one hypothesis is that the processed DEG file contains some corrupted symbols

**Comment by sinmojito** [comment]
nono it is not your fault - it is the fault of the input file you downloaded. For instance, I am trying to convert all SYMBOLs to ENTREZIDs - which works in most instances but then there always is 5 to 6% error rate. I went into the DEG file to see unusual symbols to manually cross check with the ncbi website, for instance query for 'JAG2' shows up just fine on NCBI but query for 'LOC101060175' does not. What it does is create uneven number of rows between OG file and converted file

**Comment by KristofferSandas** [comment]
So its because of the file we got from  Benjamin? In my mind it shouldnt matter if some rows are removed

**Comment by KristofferSandas** [comment]
Push it anyway, so your name shows up in the history. So far its only there once since you pushed directly to the main the first times

**Comment by sinmojito** [comment]
done

**Comment by KristofferSandas** [comment]
Ok, something messed up with the merge on my end. Can you just send me the script on whatsapp? :D

**Comment by sinmojito** [comment]
its 3:40 am and I think enrichGO is finally clicking but I'll need to make minor adjustment to your DEG file that for some reason isn't working for me. For eg, your function needs to be modified so it gives me the DEG file in excel right after you filtered them for low counts (but before padj and FDR step). I think that file is supposed to be one of the parameters of enrichGO. If you  could add an additional output along with the usual into processed data that'd be great!

**Comment by sinmojito** [comment]
*minor adjustment to your DEG code

**Comment by sinmojito** [comment]
I created a pull request where I modified your file to give me the input I want but it didnt work - u can see and make minor adjustment there if u want to preserve your OG file
